pallets/msa/src/lib.rs

```diff
@@ -1371,7 +1389,7 @@ impl<T: Config> DelegationValidator for Pallet<T> {
 	}
 }
 
-impl<T: Config> SchemaGrantValidator for Pallet<T> {
+impl<T: Config> SchemaGrantValidator<T::BlockNumber> for Pallet<T> {
 	/// Check if provider is allowed to publish for a given schema_id for a given delegator
 	///
 	/// # Errors
```

Comment:
We should bubble up the new error here as well.

Reply:
looks like `ensure_valid_delegation` this one too.